openfisca_us/variables/demographic/geographic/state_code.py

```diff
@@ -66,7 +66,81 @@ class State(Enum):
     AP = "Armed Forces Pacific"
 
 
+class StateCode(Enum):
+    AL = "AL"
+    AK = "AK"
+    AZ = "AZ"
+    AR = "AR"
+    CA = "CA"
+    CO = "CO"
+    CT = "CT"
+    DE = "DE"
+    FL = "FL"
+    GA = "GA"
+    HI = "HI"
+    ID = "ID"
+    IL = "IL"
+    IN = "IN"
+    IA = "IA"
+    KS = "KS"
+    KY = "KY"
+    LA = "LA"
+    ME = "ME"
+    MD = "MD"
+    MA = "MA"
+    MI = "MI"
+    MN = "MN"
+    MS = "MS"
+    MO = "MO"
+    MT = "MT"
+    NE = "NE"
+    NV = "NV"
+    NH = "NH"
+    NJ = "NJ"
+    NM = "NM"
+    NY = "NY"
+    NC = "NC"
+    ND = "ND"
+    OH = "OH"
+    OK = "OK"
+    OR = "OR"
+    PA = "PA"
+    RI = "RI"
+    SC = "SC"
+    SD = "SD"
+    TN = "TN"
+    TX = "TX"
+    UT = "UT"
+    VT = "VT"
+    VA = "VA"
+    WA = "WA"
+    WV = "WV"
+    WI = "WI"
+    WY = "WY"
+    DC = "DC"
+    GU = "GU"
+    MP = "MP"
+    PW = "PW"
+    PR = "PR"
+    VI = "VI"
+    AA = "AA"
+    AE = "AE"
+    AP = "AP"
+
+
 class state_code(Variable):
+    value_type = Enum
+    possible_values = StateCode
+    default_value = StateCode.CA
+    entity = Household
+    label = "State code"
+    definition_period = ETERNITY
+
+    def formula(household, period, parameters):
+        return StateCode.encode(household("state", period).decode_to_str())
+
+
+class state(Variable):
```

Comment:
Will a variable named `state` ever create problems given it can be reserved?

In general I'd prefer more descriptiveness, e.g. to indicate it's the two-letter code.

Reply:
Sure, so maybe we just need the state_code variable. Or I could rename this to state_name?

Reply:
state_name sounds good. Could you put these in separate files?